src/internal/packager/images/progress.go

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package images provides image-related utilities for Zarf
+package images
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/zarf-dev/zarf/src/pkg/utils"
+	"oras.land/oras-go/v2"
+)
+
+// Report defines a function to log progress
+type Report func(bytesRead, totalBytes int64)
+
+// DefaultReport returns a default report function
+func DefaultReport(l *slog.Logger, msg string, imageName string) Report {
+	return func(bytesRead, totalBytes int64) {
+		percentComplete := float64(bytesRead) / float64(totalBytes) * 100
```

Comment:
Minor - it may be worthwhile to check that `totalBytes` is not zero and what to do in that situation

Reply:
Fair enough, added a warn + return